pepper-apis/src/main/java/org/broadinstitute/ddp/filter/UserAuthCheckFilter.java

```diff
@@ -130,12 +130,13 @@ private void handleTemporaryUser(Request request) {
                     + " temp-user whitelisted path without a token", tempUserGuid);
             canAccess = false;
         } else if (tempUser.isExpired()) {
-            LOG.error("Temporary user with guid '{}' had already expired at time {}ms", tempUserGuid, tempUser.getExpiresAtMillis());
+            LOG.warn("Temporary user with guid '{}' had already expired at time {}ms", tempUserGuid, tempUser.getExpiresAtMillis());
             canAccess = false;
         }
 
         if (!canAccess) {
-            throw halt(401);
+            ApiError apiError = new ApiError(ErrorCodes.AUTH_CANNOT_BE_DETERMINED, "Invalid temporary user");
```

Comment:
I'm still a little hesitant about this. Note that this auth filter is used for BOTH auth users and temp users. This temp user check is a different code path and you do not need an auth token to access. Client ID and study guid is effectively public information. And our API docs are public as well. It won't be hard to get client id, study guid, and API path correct. So now, only thing you need to guess is the temp user guid. And it just so happens that we tell callers that "invalid temporary user". Whereas previously we don't say anything at all.

How about we return a payload in the regular code path but nothing in the temp user code path?

Reply:
@bskinner Is that ok with you ?

Reply:
Or return a generic message like "Unauthenticated".

Reply:
Why are we telling anyone it's a temp user?

Reply:
That works too. @ssettipalli can you make that change?